### PR TITLE
Add new api version 2024-06-01-preview for Microsoft.Cdn 

### PR DESCRIPTION
**Pull request by ZianWang02**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by ZianWang02** [comment]
The changes to the REST API definition are correcting the definition to accurately describe the service behavior, so bypass the review process by adding BreakingChange-Approved-BugFix.

1023 - TypeFormatChanged is introdued when the reviewer ask adding "format": "arm-id" in https://github.com/Azure/azure-rest-api-specs/pull/29369
1025 - RequiredStatusChange and 1034 - AddedRequiredProperty are reuired change for correcting the definition to accurately describe the service behavior in https://github.com/Azure/azure-rest-api-specs/pull/29369. The SecretVersion alway needed.
1026 - TypeChanged is to fix the definitions missing the "object" type definition.

**Comment by raosuhas** [comment]
@ZianWang02  Can you please merge your branch with the main. There seems to be some merge conflicts.

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "tunnelPolicyUpdateProperties": {
+      "properties": {
+        "tunnelType": "HttpConnect",
+        "domains": [
+          {
+            "id": "/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/customdomains/testdomain1"
+          },
+          {
+            "id": "/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/customdomains/testdomain2"
+          }
```
Comment (raosuhas) [inline specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json:17]: do you plan to add linked access checks to ensure that the caller has permissions to access these domains.

Comment (ZianWang02) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json:17]: @barathvg could you please answer this question?

**Comment by raosuhas** [comment]
Can you please take a look at the swagger lintdiff errrors and fix those as well : 

https://github.com/Azure/azure-rest-api-specs/pull/29588/checks?check_run_id=27496185108

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json (hunk 1 above)
Comment (raosuhas) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json:17]: waiting for the response here.

Comment (barathvg) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json:17]: @raosuhas yes linked access checks are added in the manifest.

Comment (barathvg) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/examples/TunnelPolicies_Update.json:17]: Here is the Snippet:
{
      "name": "profiles/tunnelpolicies",
      "commonApiVersions": [        
        "2024-06-01-preview"
      ],
      "routingType": "ProxyOnly",
      "resourceDeletionPolicy": "Cascade",
      "linkedAccessChecks": [
        {
          "actionName": "Microsoft.Cdn/profiles/tunnelpolicies/write",
          "linkedProperty": "properties.domains[*].id",
          "linkedActionVerb": "write"
        },
        {
          "actionName": "Microsoft.Cdn/profiles/tunnelpolicies/write",
          "linkedProperty": "properties.targetgroups[*].id",
          "linkedActionVerb": "write"
        }
      ],

**Comment by barathvg** [comment]
Most of these errors are from prev swagger errors, for ex, {put-responses-codes, del-response-codes, errorcontent, name patterns} are from before this change and exists on following the existing service api pattern, which if not followed will fail to adhere to existing service behaviour or lead to over-hauling of the entire service apis.

The Post-response-errors are from changes not related to http-connect. I will let @ZianWang02  comment on those.

**Comment by barathvg** [comment]
> Can you please take a look at the swagger lintdiff errrors and fix those as well :
> 
> https://github.com/Azure/azure-rest-api-specs/pull/29588/checks?check_run_id=27496185108

**Comment by ZianWang02** [comment]
> Can you please take a look at the swagger lintdiff errrors and fix those as well :
> 
> https://github.com/Azure/azure-rest-api-specs/pull/29588/checks?check_run_id=27496185108

Hi @raosuhas, because we are following our previous swagger definition and copied from old versions, can we request for exceptions for PostResponseCodes, PutResponseCodes, DeleteResponseCodes, LroErrorContent Lintdiff errors?
@barathvg For ResourceNameRestriction for profileName, can we use the already defined parameter "#/parameters/profileNameParameter" so that the pattern will automatically apply to it?
@barathvg For ResourceNameRestriction for targetGroupName and tunnelPolicyName, if we don't want to add patterns to them, can you apply for an exception with @raosuhas?

**Comment by barathvg** [comment]
> > Can you please take a look at the swagger lintdiff errrors and fix those as well :
> > https://github.com/Azure/azure-rest-api-specs/pull/29588/checks?check_run_id=27496185108
> 
> Hi @raosuhas, because we are following our previous swagger definition and copied from old versions, can we request for exceptions for PostResponseCodes, PutResponseCodes, DeleteResponseCodes, LroErrorContent Lintdiff errors? @barathvg For ResourceNameRestriction for profileName, targetGroupName and tunnelPolicyName, if we don't want to add patterns to them, can you apply for an exception with @raosuhas? Or can we use the already defined parameter "#/parameters/profileNameParameter" so that the pattern will automatically apply to it?

@ZianWang02 Thanks for pointing out the profileNameParameter that is already defined, I have fixed all the ResourceNameRestriction errors in my commit, please merge and proceed with exceptions. Thanks.

**Comment by ZianWang02** [comment]
> > > Can you please take a look at the swagger lintdiff errrors and fix those as well :
> > > https://github.com/Azure/azure-rest-api-specs/pull/29588/checks?check_run_id=27496185108
> > 
> > 
> > Hi @raosuhas, because we are following our previous swagger definition and copied from old versions, can we request for exceptions for PostResponseCodes, PutResponseCodes, DeleteResponseCodes, LroErrorContent Lintdiff errors? @barathvg For ResourceNameRestriction for profileName, targetGroupName and tunnelPolicyName, if we don't want to add patterns to them, can you apply for an exception with @raosuhas? Or can we use the already defined parameter "#/parameters/profileNameParameter" so that the pattern will automatically apply to it?
> 
> @ZianWang02 Thanks for pointing out the profileNameParameter that is already defined, I have fixed all the ResourceNameRestriction errors in my commit, please merge and proceed with remaining exceptions as requested above. Thanks.

Thank you @barathvg, I have merged the PR. @raosuhas Could you please help take another look at this PR? Thanks!

**Comment by razvanbadea-msft** [comment]
>     "DeploymentStages": {

Add also a description for this schema

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/afdx.json:7387 in 7091be1. [](commit_id = 7091be1d1ab485d190e31917632c7c123f2cf734, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>     "DeploymentStageStatus": {

Add also a description for this schema

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/afdx.json:7402 in 7091be1. [](commit_id = 7091be1d1ab485d190e31917632c7c123f2cf734, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "default": 80,

The new capacityConsciousThreshold and distributionType properties will be part of the AFDOriginGroupUpdateParameters for a PATCH opeartion. You should only have patchable properties

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/afdx.json:6761 in 7091be1. [](commit_id = 7091be1d1ab485d190e31917632c7c123f2cf734, deletion_comment = False)

**Comment by ZianWang02** [comment]
> > ```
> >       "default": 80,
> > ```
> 
> The new capacityConsciousThreshold and distributionType properties will be part of the AFDOriginGroupUpdateParameters for a PATCH opeartion. You should only have patchable properties
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2024-06-01-preview/afdx.json:6761 in [7091be1](https://github.com/Azure/azure-rest-api-specs/commit/7091be1d1ab485d190e31917632c7c123f2cf734). [](commit_id = [7091be1](https://github.com/Azure/azure-rest-api-specs/commit/7091be1d1ab485d190e31917632c7c123f2cf734), deletion_comment = False)

Hi @razvanbadea-msft , I don't quite understand why we can't add capacityConsciousThreshold or distributionType properties.
These two properties will be added in all API requests of origin group (including LIST, GET, POST, PATCH, etc.). May I know what is the patchable properties? + @anupamsr because these properties are added by him.

**Comment by razvanbadea-msft** [comment]
Because both of them has default values; you can read more about this rule here [https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-body-parameters-schema.md#why-the-rule-is-important](https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-body-parameters-schema.md#why-the-rule-is-important)

---
In reply to: [2244731342](https://github.com/Azure/azure-rest-api-specs/pull/29588#issuecomment-2244731342) [](http://example.com/codeflow?ancestors=2244731342)

**Comment by ZianWang02** [comment]
> Because both of them has default values; you can read more about this rule here https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-body-parameters-schema.md#why-the-rule-is-important
> 
> In reply to: [2244731342](https://github.com/Azure/azure-rest-api-specs/pull/29588#issuecomment-2244731342)

Fixed, thank you!

**Comment by ZianWang02** [comment]
> For other swagger lintdiff errors check the [suppression guide](https://eng.ms/docs/products/azure-developer-experience/design/specs-pr-guides/pr-suppressions)

I have added Approved-LintDiff tag according to the guide, because all the Lintdiff (PostResponseCodes, PutResponseCodes, DeleteResponseCodes, LroErrorContent, BodyTopLevelProperties) are introduced by previous version:

**Review by t-bzhan — APPROVED** [review]

**Review by ChenglongLiu — APPROVED** [review]

**Review by rkmanda — APPROVED** [review]

**Review by rkmanda — APPROVED** [review]

**Pull request merged**